src/content/SyncSavedSchedules/syncSavedSchedulesHelper.ts

```diff
@@ -0,0 +1,92 @@
+import { ISectionData } from "../App/App.types"
+
+export interface ScheduleItem {
+  text: string
+  instanceId: string
+}
+
+export async function addCoursesToSavedSchedule(
+  sections: ISectionData[],
+  scheduleID: string
+) {
+  let sectionString = ""
+  for (const section of sections) {
+    sectionString += "15194$" + section.courseID
+    if (section !== sections[sections.length - 1]) {
+      sectionString += ","
+    }
+  }
+  const urlencoded = new URLSearchParams()
+  urlencoded.append("selected", sectionString)
+  urlencoded.append("additionalParameter", scheduleID)
+
+  const requestOptions = {
+    method: "POST",
+    body: urlencoded,
+    redirect: "follow" as RequestRedirect,
+  }
+
+  return fetch(
+    "https://wd10.myworkday.com/ubc/mass-select-action/2997$20406/13710$370.htmld",
+    requestOptions
+  )
+    .then((response) => response.json())
```

Comment:
might be a good idea to use `await fetch` here, and refactor the following code to not use callbacks? they get kind of hard to follow once they're nested deeply.

Reply:
yea, the initial idea was to have just a working MVP that we can test and play with and nesting calls made debugging easier. I've went and refactored the way this function works and improved some variable naming